src/org/rust/lang/core/psi/impl/RustNamedElementImpl.kt

```diff
@@ -19,7 +19,7 @@ public abstract class RustNamedElementImpl(node: ASTNode)   : RustCompositeEleme
         throw UnsupportedOperationException();
     }
 
-    override fun getNavigationElement() = getNameElement()
+    override fun getNavigationElement() = getNameElement() ?: this
```

Comment:
Shouldn't the return type be explicitly specified here, just in case? From my (limited) understanding of Kotlin type inference, this changes `getNavigationElement` return type from `PsiElement?` to `PsiElement`.